diff --git a/node_modules/@wyw-in-js/esbuild/esm/index.mjs b/node_modules/@wyw-in-js/esbuild/esm/index.mjs
index 51afde5..e39cce7 100644
--- a/node_modules/@wyw-in-js/esbuild/esm/index.mjs
+++ b/node_modules/@wyw-in-js/esbuild/esm/index.mjs
@@ -8,12 +8,14 @@ import { readFileSync } from 'fs';
 import { basename, dirname, isAbsolute, join, parse, posix } from 'path';
 import { transformSync } from 'esbuild';
 import { slugify, transform, TransformCacheCollection, createFileReporter } from '@wyw-in-js/transform';
+import { transformSync as babelTransformSync } from '@babel/core';
 const nodeModulesRegex = /^(?:.*[\\/])?node_modules(?:[\\/].*)?$/;
 export default function wywInJS({
   debug,
   sourceMap,
   preprocessor,
   esbuildOptions,
+  babelPreTransformConfig,
   filter = /\.(js|jsx|ts|tsx)$/,
   ...rest
 } = {}) {
@@ -84,12 +86,17 @@ export default function wywInJS({
             options.jsxFragment = build.initialOptions.jsxFragment;
           }
         }
-        const transformed = transformSync(rawCode, {
+        const transformedJs = transformSync(rawCode, {
           ...options,
           sourcefile: args.path,
           sourcemap: sourceMap,
           loader
         });
+
+        const transformed = babelTransformSync(transformedJs.code,
+          babelPreTransformConfig
+        );
+
         let {
           code
         } = transformed;
